File: jobs/2026-07-03__05-39-37/encore_cron_job_cleanup_ts__t28fJhc/artifacts/user/cleanup/cleanup.ts
import { api, RawRequest, RawResponse } from "encore.dev/api";
import { CronJob } from "encore.dev/cron";
import { recordsDB } from "./records_db";

interface Record {
  id: number;
  data: string;
  created_at: string;
}

interface CreateRecordRequest {
  data: string;
  created_at?: string;
}

// POST /records – inserts a new record.
// If created_at is provided it is used, otherwise the current time is used.
export const createRecord = api(
  { method: "POST", path: "/records", expose: true },
  async (req: CreateRecordRequest): Promise<{ id: number }> => {
    const createdAt = req.created_at ?? new Date().toISOString();
    const row = await recordsDB.queryRow<{ id: number }>`
      INSERT INTO records (data, created_at)
      VALUES (${req.data}, ${createdAt})
      RETURNING id
    `;
    return { id: row!.id };
  }
);

// GET /records – returns all records ordered by id as a JSON array.
export const getRecords = api.raw(
  { method: "GET", path: "/records", expose: true },
  async (req: RawRequest, resp: RawResponse) => {
    const rows = await recordsDB.queryAll<Record>`
      SELECT id, data, created_at FROM records ORDER BY id
    `;
    resp.setHeader("Content-Type", "application/json");
    resp.end(JSON.stringify(rows));
  }
);

// POST /cleanup – deletes all records older than 1 hour.
export const cleanup = api(
  { method: "POST", path: "/cleanup", expose: true },
  async (): Promise<{ deleted: number }> => {
    const rows = await recordsDB.queryAll<{ id: number }>`
      DELETE FROM records
      WHERE created_at < NOW() - INTERVAL '1 hour'
      RETURNING id
    `;
    return { deleted: rows.length };
  }
);

// Cron job that calls the cleanup endpoint every hour.
export const _cleanupJob = new CronJob("cleanup-job", {
  title: "Cleanup stale records",
  every: "1h",
  endpoint: cleanup,
});

File: jobs/2026-07-02__17-09-42/encore_cron_job_cleanup_go__fyUasmz/artifacts/user/cleanup-app/cleanup/cleanup.go
package cleanup

import (
	"context"
	"time"

	"encore.dev/cron"
)

// CleanupResponse is the response body for POST /cleanup.
type CleanupResponse struct {
	DeletedCount int64 `json:"deleted_count"`
}

// Cleanup deletes records whose created_at is older than 24 hours.
// It is exposed as a public API so it can be invoked manually for
// verification, and it is also wired up to a Cron Job below.
//
//encore:api public method=POST path=/cleanup
func Cleanup(ctx context.Context) (*CleanupResponse, error) {
	cutoff := time.Now().Add(-24 * time.Hour)

	result, err := cleanupdb.Exec(ctx, `
		DELETE FROM records
		WHERE created_at < $1
	`, cutoff)
	if err != nil {
		return nil, err
	}

	return &CleanupResponse{DeletedCount: result.RowsAffected()}, nil
}

// CleanupCron runs every hour to remove stale records.
var _ = cron.NewJob("cleanup-old-records", cron.JobConfig{
	Title:    "Cleanup old records",
	Every:    1 * cron.Hour,
	Endpoint: Cleanup,
})